Save play time, support timer display options

Eblo opened the pull request:
Currently, save files have a timer display that shows the naive difference between the current time and the time when the file was created. This PR improves the time tracking so that it tracks the time actually spent between saves and not real time elapsed regardless of play.

2ship needs to update the tracked time in chunks with `SavingEnhancements_AdvancePlaytime()`. This gets called when:
* Saving or speaking to an owl statue
* Defeating Majora's Wrath, i.e. beating the game

Some logic for initializing and marking game completion were handled in the timer display element. I moved that to `SavingEnhancements.cpp`.

This necessitates and adds a save migration.

<!--- section:artifacts:start -->
### Build Artifacts
  - [2ship-linux.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/4649375382.zip)
  - [2ship-windows.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/4649400698.zip)
<!--- section:artifacts:end -->

balloondude2 commented:
I don't think the original timer was naive, but meant to be an RTA timer. I think we shouldn't replace it, but instead add this Play Time timer as an additional option.

Eblo commented:
Fair enough. I think giving the display different options is a good idea. I'll return to this.

Eblo commented:
I reworked it so that the timer display is not a toggle, but a list of options. Supports RTA, IGT, or no display (default).

<img width="847" height="221" alt="image" src="https://github.com/user-attachments/assets/db4e8f50-ab60-4a05-8a0d-ee87f19eb998" />

<img width="141" height="81" alt="image" src="https://github.com/user-attachments/assets/79f7f0b3-ae67-4b49-b58f-f4a1e9d9fd58" />

<img width="165" height="70" alt="image" src="https://github.com/user-attachments/assets/cfc9a8c4-1ac5-4346-ab5c-331370173d75" />

Caladius commented:
Thank you for adding this, as with https://github.com/HarbourMasters/2ship2harkinian/pull/1266 my intention here was for Speedruns specifically. The thought of just In Game Play-time was there but wasn't my priority at the time.

Eblo committed:
Save play time, support timer display options

Eblo commented:
This has been rebased against develop and is ready for re-review.

garrettjoecox commented on mm/include/z64save.h:
```diff
@@ -398,6 +398,8 @@ typedef struct ShipSaveInfo {
     SaveType saveType;
     uint64_t fileCreatedAt;
     uint64_t fileCompletedAt; // For now this is always Majora final blow, has the potential to be something else later on
+    uint64_t lastTimeLog; // Persisted value not relevant, but needs stored to calculate filePlaytime
```
See `ShipSaveContext`

garrettjoecox reviewed (approved):
Move lastTimeLog to `ShipSaveContext`, aside from that LGTM. 

Clever use of timestamps to avoid incrementing a number every frame. Both that and the ability to have both RTA & gametime should be back-ported to SoH if you're feeling adventurous.

Eblo committed:
Use ShipSaveContext, add OnGameCompletion hook

Eblo commented:
Moved the intermediate time tracking variable to `ShipSaveContext`. I noticed that completion time was not being properly tracked in Triforce Hunt, so I added a hook for `OnGameCompletion` that fires on completing the Triforce and defeating Majora. Given that this could do other things in the future, I left time splits alone, so they still set the completed time directly as before.
